README.md

```diff
@@ -1,62 +1,164 @@
 # ez-a-sync
 
-super lite docs (better coming eventually):
+## Introduction
 
+`ez-a-sync` is a Python library that enables developers to write both synchronous and asynchronous code without having to write redundant code. It provides two decorators (@a_sync('async') and @a_sync('sync')), as well as a base class (ASyncGenericBase) which can be used to create classes that can be executed in both synchronous and asynchronous contexts.
+
+## Installation
+
+`ez-a-sync` can be installed via pip:
+
+```
+pip install ez-a-sync
 ```
+
+## Usage
+
+### Decorators
+
+`ez-a-sync` provides two decorators: `@a_sync('async')` and `@a_sync('sync')`.
+
+#### `@a_sync('async')`
+
+The `@a_sync('async')` decorator can be used to define an asynchronous function that can also be executed synchronously. It should be used when the function will take longer to complete when running asynchronously.
+
+```python
 @a_sync('async')
 def some_function():
     ...
-    
-aaa = await some_function()      <- this works
-aaa = some_function(sync=True)   <- this works
 ```
 
+This function can then be executed asynchronously using `await`:
+
+```python
+aaa = await some_function()
+```
+
+It can also be executed synchronously by passing `sync=True`:
+
+```python
+aaa = some_function(sync=True)
 ```
+
+#### `@a_sync('sync')`
+
+The `@a_sync('sync')` decorator can be used to define a synchronous function that can also be executed asynchronously. It should be used when the function will take longer to complete when running synchronously.
+
+```python
 @a_sync('sync')
-async def some_async_fn():
-   ...
-   
-aaa = some_async_fn()                  <- this works
-aaa = await some_async_fn(sync=False)  <- this works
+def some_function():
+    ...
+```
+
+This function can then be executed synchronously:
+
+```python
+aaa = some_function()
 ```
 
+It can also be executed asynchronously by passing `sync=False` and using `await`:
+
+```python
+aaa = await some_function(sync=False)
 ```
+
+### Classes
+
+`ez-a-sync` also provides a base class `ASyncGenericBase` that can be used to create classes that can be executed in both synchronous and asynchronous contexts. To create an asynchronous class, simply inherit from `ASyncGenericBase` and set `asynchronous=True`:
+
+```python
 class CoolAsyncClass(ASyncGenericBase):
     asynchronous=True
     
     def some_sync_fn():
        ...   
-       
+```
+
+In this example, `CoolAsyncClass` has `asynchronous=True`, which means it is an asynchronous class. You can call `some_sync_fn` asynchronously using `await`:
+
+```python
 aaa = await CoolAsyncClass().some_sync_fn()
 ```
 
+`CoolAsyncClass` can also be called synchronously by passing `sync=True`:
+
+```python
+aaa = CoolAsyncClass().some_sync_fn(sync=True)
 ```
+
+Similarly, you can create a synchronous class by setting `asynchronous=False`:
```

Comment:
So, we should discuss this but the tldr is "pass in a flag" which can either be "sync" or "asynchronous", user's choice. Probably can swap in reference to flags here and in other places after we briefly describe them.

Reply:
```suggestion
Similarly, you can create a synchronous class by setting `sync=True` or `asynchronous=False`:
```